resilience4j-spring-boot2/src/test/java/io/github/resilience4j/circuitbreaker/monitoring/events/CircuitBreakerStreamEventsTest.java

```diff
@@ -0,0 +1,150 @@
+        dummyService.doSomething(false);
+        Thread.sleep(1000); // pause 1 sec to record the event
+        try {
+            dummyService.doSomething(true);
+        } catch (IOException ex) {
+            // Do nothing. The IOException is recorded by the CircuitBreaker as part of the recordFailurePredicate as a failure.
+        }
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
+        Thread.sleep(1000); // for webClient to complete the subscribe operation
+        assertThat (circuitBreakerEventsBefore.getCircuitBreakerEvents().size()).isLessThan(circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
+        assertThat (events.size() ).isEqualTo(2);
+    }
+
+    @Test
+    public void streamEventsbyNameAndType() throws IOException, InterruptedException {
+        List<ServerSentEvent<String>> events = getServerSentEvents(ACTUATOR_STREAM_CIRCUITBREAKER_EVENTS+ "/backendA/ERROR");
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsBefore = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
+        // The invocation is recorded by the CircuitBreaker as a success.
+        dummyService.doSomething(false);
+        Thread.sleep(1000); // to record the event
+        try {
+            dummyService.doSomething(true);
+        } catch (IOException ex) {
+            // Do nothing. The IOException is recorded by the CircuitBreaker as part of the recordFailurePredicate as a failure.
+        }
+        CircuitBreakerEventsEndpointResponse circuitBreakerEventsAfter = circuitBreakerEvents(ACTUATOR_CIRCUITBREAKEREVENTS + "/backendA");
+        Thread.sleep(1000); // for webClient to complete the subscribe operation
+        assertThat (circuitBreakerEventsBefore.getCircuitBreakerEvents().size()).isLessThan(circuitBreakerEventsAfter.getCircuitBreakerEvents().size());
+        assertThat (events.size()).isEqualTo(1);
+    }
+
+    private List<ServerSentEvent<String>> getServerSentEvents(String s) {
+        Flux<ServerSentEvent<String>> circuitBreakerStreamEventsForAfter = circuitBreakerStreamEvents(s);
+        List<ServerSentEvent<String>> events = new ArrayList<>();
+
+        circuitBreakerStreamEventsForAfter.subscribe(
+            content -> events.add(content),
+            error -> System.out.println("Error receiving SSE: {}" + error),
+            () -> System.out.println("Completed!!!"));
+        return events;
+    }
+
+    private CircuitBreakerEventsEndpointResponse circuitBreakerEvents(String s) {
+        return this.webTestClient.get().uri(s).exchange()
+            .expectStatus().isOk()
+            .expectBody(CircuitBreakerEventsEndpointResponse.class)
+            .returnResult()
+            .getResponseBody();
+    }
+
+    private Flux<ServerSentEvent<String>> circuitBreakerStreamEvents(String s) {
+        WebClient client = WebClient.create("http://localhost:" + randomServerPort);
```

Comment:
Is it possible to use the WebTestClient here?
Here is a guide how to use the WebTestClient for streaming responses: 
https://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/testing-webtestclient.pdf

Then please use the `StepVerifier` to subscribe to the stream and verify that the streams emits one event.